test/clj/rems/api/test_applications.clj

```diff
@@ -562,6 +577,39 @@
         (is (response-is-not-found?
              (api-response :get (str "/api/applications/" app-id) nil
                            api-key applicant)))))
+    (testing "can delete application with attachments"
+      (let [form-id (test-helpers/create-form! {:form/fields [{:field/id "att"
+                                                               :field/type :attachment
+                                                               :field/title {:en "x" :fi "x" :sv "x"}
+                                                               :field/optional false}]})
+            cat-id (test-helpers/create-catalogue-item! {:form-id form-id})
+            app-id (test-helpers/create-application! {:catalogue-item-ids [cat-id]
+                                                      :actor applicant})
+            att-id (-> (request :post (str "/api/applications/add-attachment?application-id=" app-id))
+                       (assoc :params {"file" filecontent})
+                       (assoc :multipart-params {"file" filecontent})
+                       (authenticate api-key applicant)
+                       handler
+                       read-ok-body
+                       :id)]
+        (assert (number? att-id))
+        (is (= {:success true}
+               (send-command applicant {:type :application.command/save-draft
+                                        :application-id app-id
+                                        :field-values [{:form form-id :field "att" :value (str att-id)}]})))
+        (is (contains? (-> (get-application-for-user app-id applicant)
+                           :application/permissions
+                           set)
+                       "application.command/delete"))
+        (is (= {:success true}
+               (api-call :post "/api/applications/delete" {:application-id app-id}
+                         api-key applicant)))
+        (is (response-is-not-found?
+             (api-response :get (str "/api/applications/" app-id) nil
+                           api-key applicant)))
+        (is (response-is-not-found?
```

Comment:
Is this duplicated?

Reply:
Oh right, I meant to check that the _attachment_ is not found, I'll fix it.